runtime/lua/vim/lsp/diagnostic.lua

```diff
@@ -720,6 +740,19 @@ function M.set_underline(diagnostics, bufnr, client_id, diagnostic_ns, opts)
       to_position(start, bufnr),
       to_position(finish, bufnr)
     )
+
+    for _, tag in pairs(diagnostic.tags or {}) do
+      local higroup_tag = diagnostictag_highlight_map[tag].group
```

Comment:
This would fail if the `diagnostictag_highlight_map` misses an entry for the tag. Probably worth protecting against that - in case a server sends some undefined tag?